lib/charliecloud.py

```diff
@@ -304,10 +306,25 @@ def layer_hashes_load(self):
       except json.JSONDecodeError as x:
          FATAL("can't parse manifest file: %s:%d: %s"
                % (self.manifest_path, x.lineno, x.msg))
-      try:
-         self.layer_hashes = [i["digest"].split(":")[1] for i in doc["layers"]]
-      except (AttributeError, KeyError, IndexError):
-         FATAL("can't parse manifest file: %s" % self.manifest_path)
+
+      schema_version = doc['schemaVersion']
+      if (schema_version == 1):
+         DEBUG('loading layer hashes from schema version 1 manifest')
+         try:
+            self.layer_hashes = [i["blobSum"].split(":")[1] for i in doc["fsLayers"]]
+         except (KeyError, AttributeError, IndexError):
+            FATAL("manifest file %s missing expected v1 keys"
```

Comment:
Could be problems other than missing key; suggest making this error message more generic. Also check whether it still needs to be wrapped to two lines.